### ISSUE TITLE
error: ‘nullptr’ was not declared in this scope 

### ISSUE THREAD

**Issue by FlorenceCa** [Itseez/opencv_contrib#659]
Hi, when I build OpenCV with the module contrib I got the following mistake : error: ‘nullptr’ was not declared in this scope (see below for details).

If someone has and idea that would be very helpful.
Thanks

```
[ 79%] Generating test_precomp.hpp.gch/opencv_test_cvv_RELEASE.gch
In file included from /net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv.hpp:19:0,
                 from /home_local/OpenCV_Contrib/opencv-3.1.0/build/modules/cvv/test_precomp.hpp:15:
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/call_meta_data.hpp: In constructor ‘cvv::impl::CallMetaData::CallMetaData()’:
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/call_meta_data.hpp:27:13: error: ‘nullptr’ was not declared in this scope
      : file(nullptr), line(0), function(nullptr), isKnown(false)
             ^
In file included from /net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv.hpp:21:0,
                 from /home_local/OpenCV_Contrib/opencv-3.1.0/build/modules/cvv/test_precomp.hpp:15:
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/dmatch.hpp: At global scope:
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/dmatch.hpp:83:47: error: ‘nullptr’ was not declared in this scope
                                const char * = nullptr, const char * = nullptr,
                                               ^
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/dmatch.hpp:83:71: error: ‘nullptr’ was not declared in this scope
                                const char * = nullptr, const char * = nullptr,
                                                                       ^
In file included from /net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv.hpp:22:0,
                 from /home_local/OpenCV_Contrib/opencv-3.1.0/build/modules/cvv/test_precomp.hpp:15:
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/filter.hpp:61:47: error: ‘nullptr’ was not declared in this scope
                                const char * = nullptr, const char * = nullptr)
                                               ^
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/filter.hpp:61:71: error: ‘nullptr’ was not declared in this scope
                                const char * = nullptr, const char * = nullptr)
                                                                       ^
In file included from /net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv.hpp:24:0,
                 from /home_local/OpenCV_Contrib/opencv-3.1.0/build/modules/cvv/test_precomp.hpp:15:
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/show_image.hpp:62:45: error: ‘nullptr’ was not declared in this scope
                              const char * = nullptr, const char * = nullptr)
                                             ^
/net/rmc-lx0075/home_local/OpenCV_Contrib/opencv_contrib-master/modules/cvv/include/opencv2/cvv/show_image.hpp:62:69: error: ‘nullptr’ was not declared in this scope
                              const char * = nullptr, const char * = nullptr)
                                                                     ^
modules/cvv/CMakeFiles/pch_Generate_opencv_test_cvv.dir/build.make:62: recipe for target 'modules/cvv/test_precomp.hpp.gch/opencv_test_cvv_RELEASE.gch' failed
make[2]: *** [modules/cvv/test_precomp.hpp.gch/opencv_test_cvv_RELEASE.gch] Error 1
CMakeFiles/Makefile2:11757: recipe for target 'modules/cvv/CMakeFiles/pch_Generate_opencv_test_cvv.dir/all' failed
make[1]: *** [modules/cvv/CMakeFiles/pch_Generate_opencv_test_cvv.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by berak** [comment]
would you be so kind and replace the screenshot with a **text** version of your errors ?

**Comment by FlorenceCa** [comment]
Done ! (but it is ugly, I don´t know how to make it more readable...)

**Comment by Dikay900** [comment]
its ok like this dont worry.
If you want to continue your build you can disable the cvv module with cmake for now.

**Comment by FlorenceCa** [comment]
How can I do that ?

**Comment by Dikay900** [comment]
should be 
`-DBUILD_OPENCV_CVV`
unfortunately i cannot confirm this since i dont have access to cmake right now but try it

**Comment by FlorenceCa** [comment]
I just try and I have the same problem

**Comment by alalek** [comment]
It is case sensitive, use `-DBUILD_opencv_cvv=OFF`

**Comment by FlorenceCa** [comment]
Hi again, 
Thank you for the answers, but I just tried once more, and I have the same issue, but I have the feeling that cvv module is not disable (during the make I still have` Scanning dependencies of target opencv_cvv` )

**Comment by TheodoreT** [comment]
I get this error as well. Disabling cvv for the moment seems to work.

**Comment by FlorenceCa** [comment]
But how did you disable cvv ? with -DBUILD_OPENCV_CVV=OFF ? Because I did this and it doesn´t change anything, like cvv was not disabled at all...

**Comment by TheodoreT** [comment]
@FlorenceCa yes, but as @alalek pointed out it is case sensitive. Therefore, it is **NOT** _-DBUILD_OPENCV_CVV=OFF_ but **-DBUILD_opencv_cvv=OFF**

**Comment by Dikay900** [comment]
changed my post accordingly in case someone else is stumbling over this thread

**Comment by FlorenceCa** [comment]
Oh ok now the installation works !!!
Thanks a lot !

**Comment by TheodoreT** [comment]
I think you should reopen this issue since, this solution is a workaround but not an actual solution ;-). cvv module still has the problem, so until someone fixes that I think that it needs to be open.

**Comment by TheodoreT** [comment]
:+1:

**Comment by mshabunin** [comment]
Please, check whether the fix from #671 works for you.

**Comment by FlorenceCa** [comment]
Yes it works !
Thank you very much.
